src/special_arrays.jl

```diff
@@ -111,6 +112,95 @@ function inverse_at!(x::AbstractVector, index, t::UnitVector, y::AbstractVector)
     index
 end
 
+####
+#### unit_vector_norm
+####
+
+struct UnitVectorNorm <: VectorTransform
+    n::Int
+    chi_prior::Bool
+    function UnitVectorNorm(n::Int; chi_prior::Bool = true)
+        @argcheck n ≥ 2 "Dimension should be at least 2."
+        new(n, chi_prior)
+    end
+end
+
+"""
+$(SIGNATURES)
+
+Transform `n ≥ 2` real numbers to a unit vector of length `n` and a radius, under the
+Euclidean norm. Returns the tuple `(normalized_vector, radius)`.
+
+When `chi_prior = true`, a prior correction is applied to the radius, which only
+affects the log Jacobian determinant. The purpose of this is to make the
+distribution proper. If you wish to use another prior, set this to `false` and use
+a manual correction, see also [`logprior`](@ref).
+
+!!! note
+    At the origin, this transform is non-bijective and non-differentiable. If
+    maximizing a target distribution whose density is constant for the unit vector,
+    then the maximizer using the Chi prior is at the origin, and behavior is undefined.
+
+!!! note
+    While ``n = 1`` would be technically possible, for practical purposes it would
+    likely suffer from numerical issues, since the transform is undefined at ``x = 0``,
+    and for a Markov chain to travel from ``y=[-1]`` to ``y=[1]``, it would have to leap
+    over the origin, which is only even possible due to discretization and likely will
+    often not work. Because of this, it is disallowed.
+"""
+unit_vector_norm(n::Int; chi_prior::Bool = true) = UnitVectorNorm(n; chi_prior)
+
+nonzero_logprior(t::UnitVectorNorm) = t.chi_prior
+
+function logprior(t::UnitVectorNorm, (y, r))
```

Comment:
Maybe as below restrict the second argument to
```suggestion
function logprior(t::UnitVectorNorm, (y, r)::Tuple{AbstractVector,Real})
```
?